autoload/asyncomplete.vim

```diff
@@ -278,9 +278,8 @@ function! s:remote_refresh(ctx, force) abort
         let l:typed_len = l:endpos - l:startpos
         if l:typed_len == 1
             call add(l:sources_to_notify, l:name)
-        elseif has_key(s:matches, l:name) && s:matches[l:name]['refresh']
-            call add(l:sources_to_notify, l:name)
-        elseif s:supports_smart_completion()
```

Comment:
`smart completion` is required when every keystroke needs to refresh the list. hence we do need it. You need lua.